sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/documentanalysis/DocumentAnalysisAsyncClient.java

```diff
@@ -329,6 +329,11 @@ public final class DocumentAnalysisAsyncClient {
                     + " be null or empty"));
             }
 
+            if (document.getLength() == null) {
+                throw logger.logExceptionAsError(new IllegalArgumentException("'document length' is required and cannot"
```

Comment:
Since it's not obvious how the length is set for BinaryData (as it's not a direct param on the method), we should link to the BinaryData documentation on how to set the length.